### PR TITLE
bsunanda:Phase2-FT07 Simulation of the Fast Timing Layers

### PR DESCRIPTION
**Pull request by bsunanda**
Extend the endcap fsat timing layer to have a larger solid angle coverage. 
Data format modified to accommodate both barrel and endcap parts
Geometry files modified to accommodate specpar parameters required for Reco geometry and numbering scheme. Accordingly the configuration is changed
Two different simhit collections for barrel and endcap part - need change in the Oscar producer
Appropriate change in handling hits in sensitive detector class
Removal of obsolete xml files for FastTiming studies

### CODE REVIEW

**Comment by bsunanda** [comment]
@lgray Please watch this PR

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by lgray** [comment]
@kpedro88 This overwrited I2 with something completely functional

@bsunanda Thanks a lot! Could you provide a few plots of sim hit multiplicities?

File: Configuration/Geometry/python/GeometryExtended2023D5_cff.py [hunk 1]
```diff
@@ -9,3 +9,5 @@
 from Geometry.HcalCommonData.hcalDDDSimConstants_cfi import *
 from Geometry.HGCalCommonData.hgcalV6ParametersInitialization_cfi import *
 from Geometry.HGCalCommonData.hgcalV6NumberingInitialization_cfi import *
+from Geometry.HGCalCommonData.fastTimeParametersInitialization_cfi import *
```
Comment (kpedro88) [inline Configuration/Geometry/python/GeometryExtended2023D5_cff.py]: these lines should be added as the "reco" entry for I2 in timingDict in dict2023Geometry.py (no direct editing of 2023 geometry configs)

**Comment by lgray** [comment]
@bsunanda I am guessing the RecoGeometry itself comes in another PR?

**Comment by lgray** [comment]
@bsunanda please address Kevin's comments

I've checked the sim hit multiplicities myself now and I am proceeding to write a digitizer.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by lgray** [comment]
urgent

File: Configuration/Geometry/scripts/dict2023Geometry.py [hunk 2]
```diff
@@ -437,16 +437,18 @@
...
+            'Geometry/HGCalCommonData/data/fastTimingElement.xml',
             ],
         3 : [
             'Geometry/HGCalSimData/data/fasttimesens.xml'
             ],
         4 : [
             'Geometry/HGCalSimData/data/fasttimeProdCuts.xml'
             ],
+        "sim" : [
+            'from Geometry.HGCalCommonData.fastTimeParametersInitialization_cfi import *'
```
Comment (kpedro88) [inline Configuration/Geometry/scripts/dict2023Geometry.py]: `from Geometry.HGCalCommonData.fastTimeNumberingInitialization_cfi import *` also needs to be added

**Comment by lgray** [comment]
@bsunanda You missed a line. Please update!

**Comment by kpedro88** [comment]
If I run `python generate2023Geometry.py -C 2 -I 2`, it should generate exactly the D5 geometry files that are tracked in CMSSW. Currently that is not the case. I recommend doing that explicitly whenever making a change to the geometry, rather than just editing the scripts after the fact.

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Configuration/Geometry/python/GeometryExtended2023D5_cff.py (hunk 1 above)
Comment (kpedro88) [inline Configuration/Geometry/python/GeometryExtended2023D5_cff.py]: Yeah, that was my mistake. It's correct in the new commits.

**Comment by bsunanda** [comment]
I shall be preparing a simple analyzer to make a few validation plots. Later on they can be made into step 1 of validation toolkit
________________________________

**Comment by lgray** [comment]
@cmsbuild please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by davidlange6** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by lgray** [comment]
@civanch please take a look! Thanks!

**Review by civanch — COMMENTED** [review]
@bsunanda , why this producer use edm::ESProducer interface and not edm::one:: or edm::stream ?

Why it returns auto_ptr , which is forbidden now?

**Comment by lgray** [comment]
@civanch There is no such thing as an ```edm::one::``` or ```edm::stream::``` ESProducer, and it has not been enforced to use unique_ptr yet as far as I know. What you are talking about only applies to EDProducers.

**Comment by civanch** [comment]
@lgray , concerning ESProducer may be I wrong but auto_ptr should be substituted - there was a campaign not long ago over CMSSW.

But more warring me Geometry/HGCalCommonData/BuildFile.xml  where a dependence on Geant4 is introduced. I suspect it is not optimal.

**Comment by Dr15Jones** [comment]
The use of `std::shared_ptr` is the preferred return type from an `ESProducer`. `std::auto_ptr` should be avoided in all new code.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@civanch To avoid dependence on Geant4 some of the service functions need to be redefined.

**Comment by bsunanda** [comment]
@civanch Do you really want the changes in the service functions - I prefer to keep a call with local position rather the explicit values of x, y, z or some other derived quantities

**Comment by bsunanda** [comment]
@ianna @civanch Could you please approve this PR?

**Comment by ianna** [comment]
@bsunanda - please, remove G4 dependency from HGCalCommonData package (see my comment above). Geometry packages should not depend on G4. It will introduce cyclic dependencies.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna @civanch Could you sign this PR now - having gotten rid of G4 dependency in Geometry/HGCalCommonData

**Comment by lgray** [comment]
@bsunanda I have found a bug. In the detid the iphi() only ever goes to 256, which seems suspiciously artificial.

This screen shot is from minbias, and as you can see the full bitwidth is not spanned. Please take a look.
screen shot 2016-10-27 at 12 57 50 pm

**Comment by lgray** [comment]
Indeed, if you look at the detId class I think the masks for izeta() and iphi() are switched!

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
@bsunanda you still have `kFastTimeCellZMask       = 0x3FF;`, I don't think that is right.

**Comment by bsunanda** [comment]
@kpedro88 we have 10 bits for both z|eta as well as phi. This can fit in 32 bit

**Comment by kpedro88** [comment]
Ah, I see the offsets are consistent. Thanks

**Comment by lgray** [comment]
@kpedro88 Nah, that's fine.

**Comment by bsunanda** [comment]
@ianna @civanch Could you please approve this now?

**Comment by ianna** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**